resources/definition.rb

```diff
@@ -30,21 +48,40 @@
     package package_deps
   end
 
-  definition_path = ::File.join(new_resource.prefix_path, new_resource.definition)
+  installation_path = if new_resource.version_prefix
+                        ::File.join(new_resource.prefix_path, new_resource.definition)
+                      else
+                        new_resource.prefix_path
+                      end
 
-  if platform_family?('mac_os_x')
-    env = { 'RUBY_CONFIGURE_OPTS' => "--with-openssl-dir=#{openssl_prefix.to_s}" }
-    env.merge(new_resource.environment) unless new_resource.environment.nil?
-  else
-    env = new_resource.environment
+  env = if platform_family?('mac_os_x')
+          extra_env = { 'RUBY_CONFIGURE_OPTS' => "--with-openssl-dir=#{openssl_prefix}" }
+          new_resource.environment.merge extra_env
+        else
+          new_resource.environment
+        end
+
+  ruby_build_cmd = [
+    '/usr/local/bin/ruby-build',
+    new_resource.definition,
+    installation_path,
+  ].join(' ')
+
+  if new_resource.patch
+    patch_path = "#{Chef::Config[:file_cache_path]}/#{new_resource.patch}"
+    ruby_build_cmd += %( --patch < "#{patch_path}" )
+
+    cookbook_file patch_path do
+      source new_resource.patch
+    end.run_action(:create)
```

Comment:
We really don't want to be using these `run_action` anymore to force it into compile time. I'd be loathed to include it in a cookbook, as we'll be removing it really soon. 

Another way of saying it: I don't think this this is the right path to take right now

Reply:
I wondered about that. I wasn't completely sure why it was being forced into compile time, but if there isn't a good reason to do that, it probably shouldn't be there. I'll take it out.

Reply:
Reading your follow-up comment about compile time: I didn't add that. It was already there, and I didn't feel confident enough to take it out. It's gone from this PR now, though.